### PR TITLE
Fix output tensor dimension in degenerate `optimize_acqf` call

### LINKED ISSUE #2740: [Bug]: Inconsistent output dimensions
### What happened?

Not sure if this is intentional, but it seems like a bug to me. When fixing all features in `optimize_acqf`, the returned acquisition value changes from a float to a 1-D tensor. Haven't checked what happens for the other `optimize_*` functions.

Of course, this is a degenerate case since calling the optimization routing with zero degrees of freedom is sort of meaningless, but it can still be useful when invoking BoTorch procedurally from wrappers as in my case.

### Please provide a minimal, reproducible example of the unexpected behavior.

Here a minimal example adjusted from the landing page code:

```python
import torch
from botorch.acquisition import LogExpectedImprovement
from botorch.fit import fit_gpytorch_mll
from botorch.models import SingleTaskGP
from botorch.models.transforms import Normalize, Standardize
from botorch.optim import optimize_acqf
from gpytorch.mlls import ExactMarginalLogLikelihood

def run(fixed_features):

    train_X = torch.rand(10, 2, dtype=torch.double) * 2
    Y = 1 - torch.linalg.norm(train_X - 0.5, dim=-1, keepdim=True)
    Y = Y + 0.1 * torch.randn_like(Y)

    gp = SingleTaskGP(
        train_X=train_X,
        train_Y=Y,
        input_transform=Normalize(d=2),
        outcome_transform=Standardize(m=1),
    )
    mll = ExactMarginalLogLikelihood(gp.likelihood, gp)
    fit_gpytorch_mll(mll)
    logEI = LogExpectedImprovement(model=gp, best_f=Y.max())
    bounds = torch.stack([torch.zeros(2), torch.ones(2)]).to(torch.double)
    candidate, acq_value = optimize_acqf(
        logEI,
        bounds=bounds,
        q=1,
        num_restarts=5,
        raw_samples=20,
        fixed_features=fixed_features,
    )
    print(acq_value)

run({0: 0})  
run({0: 0, 1: 0})
```

### Please paste any relevant traceback/logs produced by the example provided.

```shell
tensor(-4.7173, dtype=torch.float64)
tensor([-8.1727], dtype=torch.float64)
```

### BoTorch Version

0.13.0

### Python Version

3.10

### Operating System

macOS

### Code of Conduct

- [x] I agree to follow BoTorch's Code of Conduct

### PR DESCRIPTION
**Pull request by AdrianSosic**
## Motivation

Fixes #2740 

@sdaulton: This is a draft for fixing the dimensionality issue. While the PR technically solves the problem, think it only addresses the symptom and not the underlying root cause. For the latter, I'd need some guidance from your end. 

Specifically, something seems fishy about the tensor dimensions promised by the [acquisition function base class](https://github.com/pytorch/botorch/blob/d74a11331c2efb89c9b5ea688d2a61c1d9fb1420/botorch/acquisition/acquisition.py#L64-L73). The docstring states that the input is a `(b) x q x d`-dim tensor and the output is a `(b)`-dim. The way I understand this is that `b` is an optional batching dimension. But if you read the contract, it would imply:
* If you pass a `b x q x d`-dim tensor, you get a `(b)`-dim tensor back.
* If you pass a `q x d`-dim tensor, you get a dimensionless tensor (i.e. scalar value) back.

Now the problem is that the second part seems to be actually violated, which causes the the problem in #2740. Interestingly, when writing the test, I wanted to use a `MockAcquisitionFunction()` object for it like done in other tests. However, that object *does seem* to fulfill the contract, so I couldn't even reproduce the problem with the mock but had to reuse the code from my original minimal example.

Any thoughts how we should go about this, i.e. is this actually a more fundamental problem in the acquisition function code?

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

A corresponding test has been included.

### CODE REVIEW

**Comment by sdaulton** [comment]
Thanks for the PR! It looks like a test is failing. Would you mind fixing that please?

**Comment by AdrianSosic** [comment]
> Thanks for the PR! It looks like a test is failing. Would you mind fixing that please?

Sorry, my bad, forgot to run the entire test suite. Should now be fixed – though there are 4 other tests failing also on the main branch?

Still, like I mentioned earlier: while the change fixes the particular problem reported, I don't think it properly addresses the root cause. So input would be welcome 🙃
